osu.Game/Online/Rooms/PlaylistExtensions.cs

```diff
@@ -38,7 +39,21 @@ public static IEnumerable<PlaylistItem> GetUpcomingItems(this IEnumerable<Playli
                 : GetUpcomingItems(playlist).First();
         }
 
+        /// <summary>
+        /// Returns the total duration from the <see cref="PlaylistItem"/> in playlist order from the supplied <paramref name="playlist"/>,
+        /// </summary>
         public static string GetTotalDuration(this BindableList<PlaylistItem> playlist) =>
-            playlist.Select(p => p.Beatmap.Length).Sum().Milliseconds().Humanize(minUnit: TimeUnit.Second, maxUnit: TimeUnit.Hour, precision: 2);
+            playlist.Select(p =>
+            {
+                double rate = 1;
+
+                if (p.RequiredMods.Length > 0)
+                {
+                    var ruleset = p.Beatmap.Ruleset.CreateInstance();
+                    rate = ModUtils.CalculateRateWithMods(p.RequiredMods.Select(mod => mod.ToMod(ruleset)));
```

Comment:
This isn't going to work for wind up/down but I guess this is better than nothing for now.